subsys/net/lib/azure_iot_hub/src/azure_iot_hub_dps.c

```diff
@@ -91,54 +131,127 @@ static int dps_settings_handler(const char *key, size_t len,
 				settings_read_cb read_cb, void *cb_arg)
 {
 	int err;
+	static uint32_t hostname_len;
+	static uint32_t device_id_len;
```

Comment:
Why are they static?

Reply:
It looks like each value stored to settings are retrieved separately in multiple calls to `dps_settings_handler`. So the lengths of the hostname and device ID buffers needs to be maintained in between calls.